src/test/store/receive-profile.test.js

```diff
@@ -762,7 +763,10 @@ describe('actions/receive-profile', function () {
             'web-channel': setupWithWebChannel,
           }[setupWith];
           const { dispatch, getState } = setupFn(profileAs);
-          await dispatch(retrieveProfileFromBrowser());
+          const browserConnectionStatus = await createBrowserConnection(
+            'Firefox/123.0'
+          );
```

Comment:
Shouldn't that be in `setupWithWebChannel`? It would also make that `setupWithFrameScript` doesn't have it for example, which I think makes sense.

Reply:
> Shouldn't that be in `setupWithWebChannel`?

Putting it into `setupWithWebChannel` would mean that the setup function becomes asynchronous, and then we have to add a million `await`s. In the future I'd like to simplify this test to create its own `BrowserConnection` object, synchhronously. This can then be done in the setup function. And we can test `createBrowserConnection` in its own test file, without having to deal with its asynchronicity in the other tests.

> It would also make that `setupWithFrameScript` doesn't have it for example, which I think makes sense.

The frame script path also checks for the WebChannel. We no longer support old versions of Firefox which only have the frame script and no WebChannel at all. We require at least the "basic" WebChannel.